core/src/main/java/com/datastrato/gravitino/storage/relation/RelationEntityStore.java

```diff
@@ -0,0 +1,120 @@
+  private RelationBackend backend;
+
+  @Override
+  public void initialize(Config config) throws RuntimeException {
+    this.backend = createRelationEntityBackend(config);
+  }
+
+  private static RelationBackend createRelationEntityBackend(Config config) {
+    String backendName = config.get(ENTITY_RELATION_STORE);
+    String className = RELATION_BACKENDS.getOrDefault(backendName, backendName);
+    if (Objects.isNull(className)) {
+      throw new RuntimeException("Unsupported backend type..." + backendName);
+    }
+
+    try {
+      RelationBackend relationBackend =
+          (RelationBackend) Class.forName(className).getDeclaredConstructor().newInstance();
+      relationBackend.initialize(config);
+      return relationBackend;
+    } catch (Exception e) {
+      LOGGER.error("Failed to create and initialize RelationBackend by name '{}'.", backendName, e);
+      throw new RuntimeException(
+          "Failed to create and initialize RelationBackend by name: " + backendName, e);
+    }
+  }
+
+  @Override
+  public void setSerDe(EntitySerDe entitySerDe) {
+    throw new UnsupportedOperationException("Unsupported operation in relation entity store.");
+  }
+
+  @Override
+  public <E extends Entity & HasIdentifier> List<E> list(
+      Namespace namespace, Class<E> type, Entity.EntityType entityType) throws IOException {
+    return backend.list(namespace, entityType);
+  }
+
+  @Override
+  public boolean exists(NameIdentifier ident, Entity.EntityType entityType) throws IOException {
+    return backend.exists(ident, entityType);
+  }
+
+  @Override
+  public <E extends Entity & HasIdentifier> void put(E e, boolean overwritten)
+      throws IOException, EntityAlreadyExistsException {
+    backend.insert(e, overwritten);
+  }
+
+  @Override
+  public <E extends Entity & HasIdentifier> E update(
+      NameIdentifier ident, Class<E> type, Entity.EntityType entityType, Function<E, E> updater)
+      throws IOException, NoSuchEntityException, AlreadyExistsException {
+    return backend.update(ident, entityType, updater);
+  }
+
+  @Override
+  public <E extends Entity & HasIdentifier> E get(
+      NameIdentifier ident, Entity.EntityType entityType, Class<E> e)
+      throws NoSuchEntityException, IOException {
+    E entity = backend.get(ident, entityType);
+    if (entity == null) {
+      throw new NoSuchEntityException(ident.toString());
+    }
+    return entity;
+  }
+
+  @Override
+  public boolean delete(NameIdentifier ident, Entity.EntityType entityType, boolean cascade)
+      throws IOException {
+    return backend.delete(ident, entityType, cascade);
+  }
+
+  @Override
+  public <R, E extends Exception> R executeInTransaction(Executable<R, E> executable) {
+    throw new UnsupportedOperationException("Unsupported operation in relation entity store.");
+  }
+
+  @Override
+  public void close() throws IOException {
+    backend.close();
+  }
+}
```

Comment:
I'm wondering if it is possible to add UTs for this class, we can use mockito to mock a relational backend.

Reply:
Added some simple UTs, will add more detailed UTs in the next PR